src/ui/PrintableUri/PrintableUri.ts

```diff
@@ -113,23 +116,44 @@ export class PrintableUri extends Component {
     return link.element;
   }
 
-  private renderParentsXml(element: HTMLElement, parentsXml: string) {
+  private renderParentsXml(element: HTMLElement, parentsXml: string, startAt = 0, partsCount = DeviceUtils.isMobileDevice() ? 3 : 5) {
+    $$(element).empty();
     const xmlDoc: XMLDocument = Utils.parseXml(parentsXml);
     const parents = xmlDoc.getElementsByTagName('parent');
-    const tokens: HTMLElement[] = [];
-    const separators: HTMLElement[] = [];
-    for (let i = 0; i < parents.length; i++) {
-      if (i > 0) {
-        const separator = this.buildSeparator();
-        separators.push(separator);
-        element.appendChild(separator);
+    if (startAt > 0) {
+      this.appendEllipsis(element, parentsXml, Math.max(0, startAt - partsCount + 1), partsCount);
+      this.appendSeparator(element);
+    }
+    let lastIndex = Math.min(parents.length - 2, startAt + partsCount - 2);
```

Comment:
I'd extract the `2` to a meaningful variable, I see it used twice here and once at line no134.
Also not sure if "lastIndex" is clear enough here 🤔 could say it's the last index of ...

Reply:
Good point, I'll split this across multiple variables for readability:
* `lastIndex`
* `lastIndexExcludingLastPart`
* `lastIndexToDisplay`